From: Detlev Casanova <detlev.casanova@collabora.com>
Date: Fri, 20 Jan 2023 14:32:18 -0500
Subject: Add vendor specific changes

Vendor kernel seems to be based on the following commit version of the
rockchip kernel: commit de9477630d9c6c15d7b50b0154a1d2fbfb430885 on
which some extra changes have been applied.

Those changes are necessary for the DSI cameras to be usable.

Signed-off-by: Detlev Casanova <detlev.casanova@collabora.com>
---
 .../wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile  |  4 +-
 drivers/regulator/core.c                           | 14 ++--
 3 files changed, 39 insertions(+), 59 deletions(-)

--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile
@@ -31,7 +31,7 @@
 #CONFIG_BCMDHD_CUSB := y
 #CONFIG_BCMDHD_NO_POWER_OFF := y
 CONFIG_BCMDHD_PROPTXSTATUS := y
-CONFIG_DHD_USE_STATIC_BUF := y
+# CONFIG_DHD_USE_STATIC_BUF := y
 CONFIG_BCMDHD_STATIC_BUF_IN_DHD := y
 #CONFIG_BCMDHD_ANDROID_VERSION := 14
 CONFIG_BCMDHD_AUTO_SELECT := y
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -1913,12 +1913,14 @@
 		regulator->dev = dev;
 
 		/* Add a link to the device sysfs entry */
-		err = sysfs_create_link_nowarn(&rdev->dev.kobj, &dev->kobj,
-					       supply_name);
-		if (err) {
-			rdev_dbg(rdev, "could not add device link %s: %pe\n",
-				  dev->kobj.name, ERR_PTR(err));
-			/* non-fatal */
+		if (device_is_registered(dev)) {
+			err = sysfs_create_link_nowarn(&rdev->dev.kobj, &dev->kobj,
+						       supply_name);
+			if (err) {
+				rdev_dbg(rdev, "could not add device link %s: %pe\n",
+					dev->kobj.name, ERR_PTR(err));
+				/* non-fatal */
+			}
 		}
 	}
 
